tests/test_client.py

```diff
@@ -27,312 +26,469 @@ def get_doc(
+    rec_2 = res_2.json
 
     data = get_doc()
     data["uploader"] = "uploader_123"
-    r3 = swg_index_client.add_entry(data)
-
-    r = swg_index_client.list_entries(uploader="uploader_123")
-    assert len(r.records) == 2
-    assert {r2.did, r3.did} == {r.records[0].did, r.records[1].did}
-    assert r.records[0].uploader == "uploader_123"
-    assert r.records[1].uploader == "uploader_123"
-
-
-def test_list_entries_with_uploader_wrong_uploader(swg_index_client):
+    res_3 = client.post("/index/", json=data, headers=user)
+    assert res_3.status_code == 200
+    rec_3 = res_3.json
+
+    data_grab = client.get("/index/?uploader=uploader_123")
+    assert data_grab.status_code == 200
+    data_list = data_grab.json
+    assert len(data_list["records"]) == 2
+    ids = {record["did"] for record in data_list["records"]}
+    assert len(ids) == 2
+    assert rec_2["did"] in ids
+    assert rec_3["did"] in ids
+    assert data_list["records"][0]["uploader"] == "uploader_123"
+    assert data_list["records"][1]["uploader"] == "uploader_123"
+
+
+def test_list_entries_with_uploader_wrong_uploader(client, user):
     """
     Test that returns no record due to wrong uploader id
     """
     data = get_doc()
     data["uploader"] = "uploader_1"
-    r = swg_index_client.add_entry(data)
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
 
     data = get_doc()
     data["uploader"] = "uploader_123"
-    r = swg_index_client.add_entry(data)
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
 
     data = get_doc()
     data["uploader"] = "uploader_123"
-    r = swg_index_client.add_entry(data)
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
 
-    r = swg_index_client.list_entries(uploader="wrong_uploader")
-    assert len(r.records) == 0
+    data_grab = client.get("/index/?uploader=wrong_uploader")
+    data_list = data_grab.json
+    assert len(data_list["records"]) == 0
 
 
-def test_create_blank_record(swg_index_client):
+def test_create_blank_record(client, user):
     """
     Test that new blank records only contain the uploader
     and optionally file_name fields: test without file name
     """
 
     doc = {"uploader": "uploader_123"}
-    r = swg_index_client.create_blank_entry(doc)
-    assert r.did
-    assert r.rev
-    assert r.baseid
+    res = client.post("/index/blank/", json=doc, headers=user)
+    assert res.status_code == 201
+    rec = res.json
+    assert rec["did"]
+    assert rec["rev"]
+    assert rec["baseid"]
+
+    res = client.get("/index/?uploader=uploader_123")
+    assert res.status_code == 200
+    rec = res.json
+    assert rec["records"][0]["uploader"] == "uploader_123"
+    assert not rec["records"][0]["file_name"]
 
-    r = swg_index_client.list_entries(uploader="uploader_123")
-    assert r.records[0].uploader == "uploader_123"
-    assert not r.records[0].file_name
```

Comment:
these 2 lines should not have been deleted

Reply:
Are these two not mirrored by the two lines I added above?

Reply:
ha, i missed that, sorry :p

Reply:
no worries!